### PR TITLE
Remove Canvas.setMatrix and MaskFilter flags

### PR DESCRIPTION
**Pull request by abarth**
The Skia team says these APIs are outdated and should be removed. They
do not appear to have any clients.

Fixes https://github.com/flutter/flutter/issues/7202
Fixes https://github.com/flutter/flutter/issues/7204

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @chinmaygarde

File: lib/ui/painting.dart [hunk 1]
```diff
@@ -857,23 +857,10 @@ class MaskFilter extends NativeFieldWrapperClass2 {
   ///
   /// Even in the best conditions and with the lowest quality settings, a blur
   /// is an expensive operation and blurs should therefore be used sparingly.
-  MaskFilter.blur(BlurStyle style, double sigma, {
```
Comment (Hixie) [inline lib/ui/painting.dart]: we should also update the docs. :-)

**Comment by Hixie** [comment]
LGTM despite not having tests, because it's removing code.

Please update the docs first though. They go into some lengths about these arguments.

File: lib/ui/painting.dart (hunk 1 above)
Comment (abarth) [inline lib/ui/painting.dart]: Done.

**Pull request merged**